src/forge/actors/generator.py

```diff
@@ -596,20 +596,20 @@ async def update_weights(self, version: int) -> None:
         t.stop()
 
     @endpoint
-    async def _test_save_model_params(self):
-        """Save model parameters before weight update, used for tesing purposes only."""
-        logger.info("[GeneratorWorker] save model parameters for testing.")
+    async def save_model_params(self):
+        """Used for debugging purposes. Save model parameters before weight update."""
+        logger.info("[GeneratorWorker] save model parameters for debugging.")
         for name, param in self.worker.model_runner.model.named_parameters():
-            self._test_prev_params[name] = param.detach().cpu()
+            self.debug_saved_params[name] = param.detach().cpu()
         logger.info(
             "[GeneratorWorker] finished saving model parameters, len = %d",
-            len(self._test_prev_params),
+            len(self.debug_saved_params),
         )
 
     @endpoint
-    async def _test_validate_model_params(self, validate_fn):
-        """Validate updated model params using validate_fn."""
-        logger.info("[GeneratorWorker] start validating model parameters.")
+    async def validate_model_params(self, validate_fn):
```

Comment:
Actually - could we expose an endpoint to get the underlying model? Then these methods could be entirely deleted, no?

Sorry if I'm misunderstanding. But I don't like having a debug param saved on this model :/

Reply:
> > saves the cost of message SerDer.
> 
> Do we imagine testing w/ these methods at a substantial scale?

My hunch is so long as we're still iterating on weight sync strategy, having some capability to poke into the states across multiple workers under multi-node, multi-replica run might be super valuable.